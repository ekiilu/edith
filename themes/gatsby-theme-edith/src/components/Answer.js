import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Card from "@material-ui/core/Card"
import CardActionArea from "@material-ui/core/CardActionArea"
import CardContent from "@material-ui/core/CardContent"
import Typography from "@material-ui/core/Typography"

const useStyles = makeStyles(theme => ({
  root: {
    backgroundColor: theme.palette.background.default,
  },
  colorCorrect: {
    backgroundColor: theme.palette.success.main,
  },
  colorIncorrect: {
    backgroundColor: theme.palette.error.main,
  },
  colorCorrectDisabled: {
    backgroundColor: theme.palette.success.main,
  },
  colorIncorrectDisabled: {
    backgroundColor: theme.palette.error.main,
  },
  disabled: {
    color: theme.palette.text.disabled,
  },
}))

export default props => {
  const classes = useStyles()

  let className
  if (props.color === "correct") {
    className = props.disabled
      ? classes.colorCorrectDisabled
      : classes.colorCorrect
  } else if (props.color === "incorrect") {
    className = props.disabled
      ? classes.colorIncorrectDisabled
      : classes.colorIncorrect
  } else {
    className = classes.root
  }

  return (
    <Card className={className}>
      <CardActionArea onClick={props.onClick} disabled={props.disabled}>
        <CardContent>
          <Typography className={props.disabled ? classes.disabled : null}>
            {props.label}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  )
}
