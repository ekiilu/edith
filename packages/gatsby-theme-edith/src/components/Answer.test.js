import React from "react"
import { render } from "@testing-library/react"

import Answer from "./Answer"

describe("Answer", () => {
  it("renders correctly by default", () => {
    const { asFragment } = render(<Answer label="Foo" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly when answer is correct", () => {
    const { asFragment } = render(<Answer label="Foo" color="correct" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly when answer is incorrect", () => {
    const { asFragment } = render(<Answer label="Foo" color="incorrect" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly when disabled", () => {
    const { asFragment } = render(<Answer label="Foo" disabled />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly when disabled and answer is correct", () => {
    const { asFragment } = render(
      <Answer label="Foo" color="correct" disabled />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly when disabled and answer is incorrect", () => {
    const { asFragment } = render(
      <Answer label="Foo" color="incorrect" disabled />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
