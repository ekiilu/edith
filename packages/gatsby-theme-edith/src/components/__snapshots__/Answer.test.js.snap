// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer renders correctly by default 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Answer renders correctly when answer is correct 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-colorCorrect-84 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Answer renders correctly when answer is incorrect 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-colorIncorrect-167 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Answer renders correctly when disabled 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-247 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-disabled-252 MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Answer renders correctly when disabled and answer is correct 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-colorCorrectDisabled-322 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-disabled-324 MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Answer renders correctly when disabled and answer is incorrect 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-colorIncorrectDisabled-395 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root Mui-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root makeStyles-disabled-396 MuiTypography-body1"
        >
          Foo
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</DocumentFragment>
`;
